### PR TITLE
[Merge-Queue] Add step to make comment on pull requests

### PR DESCRIPTION
**Pull request by JonWBedard**
#### 7b036cde62860bbb0f5ad2bf5c41153423601079
<pre>
[Merge-Queue] Add step to make comment on pull requests
<a href="https://bugs.webkit.org/show_bug.cgi?id=237962">https://bugs.webkit.org/show_bug.cgi?id=237962</a>
&lt;rdar://problem/90371727 &gt;

Reviewed by Aakash Jain.

* Tools/CISupport/ews-build/steps.py:
(GitHubMixin.comment_on_pr): Make a comment on pull request for in repository.
(ApplyPatch.evaluateCommand): Rename CommentOnBug to LeaveComment.
(ValidateCommiterAndReviewer.fail_build): Ditto.
(ValidateChangeLogAndReviewer.evaluateCommand): Ditto.
(LeaveComment): Renamed from CommentOnBug, support pull request
(LeaveComment.start):
(LeaveComment.getResultSummary):
(AnalyzeCompileWebKitResults.analyzeResults): Rename CommentOnBug to LeaveComment.
(AnalyzeLayoutTestsResults.report_failure): Ditto.
(FindModifiedChangeLogs.evaluateCommand): Ditto.
(CreateLocalGITCommit.evaluateCommand): Ditto.
(PushCommitToWebKitRepo.evaluateCommand): Ditto.
(CommentOnBug): Renamed to LeaveComment.

Canonical link: <a href="https://commits.webkit.org/248505@main">https://commits.webkit.org/248505@main</a>
git-svn-id: <a href="https://svn.webkit.org/repository/webkit/trunk@291373">https://svn.webkit.org/repository/webkit/trunk@291373</a> 268f45cc-cd09-0410-ab3c-d52691b4dbfc
</pre>

### CODE REVIEW

**Comment by JonWBedard** [comment]
An example of us using this: https://ews-build.webkit-uat.org/#/builders/85/builds/50

File: Tools/CISupport/ews-build/steps.py [hunk 1]
```diff
@@ -259,6 +259,29 @@ def modify_label(self, pr_number, label, repository_url=None, action='add'):
             return False
         return True
 
+    def make_comment(self, pr_number, content, repository_url=None):
```
Comment (aj062) [inline Tools/CISupport/ews-build/steps.py]: make_comment => comment_on_pr (to be consistent with comment_on_bug)

File: Tools/CISupport/ews-build/steps.py [hunk 2]
```diff
@@ -1642,6 +1665,37 @@ def doStepIf(self, step):
         return CURRENT_HOSTNAME == EWS_BUILD_HOSTNAME
 
 
+class CommentOnPullRequest(buildstep.BuildStep, GitHubMixin):
```
Comment (aj062) [inline Tools/CISupport/ews-build/steps.py]: Why not just combine CommentOnBug and this one together?

Comment (JonWBedard) [reply Tools/CISupport/ews-build/steps.py]: I can do that. What would such a class be called? `CommentOn`?

Comment (aj062) [reply Tools/CISupport/ews-build/steps.py]: not sure, maybe `CommentOnChange` or `LeaveComment`

File: Tools/CISupport/ews-build/steps.py [hunk 3]
```diff
@@ -259,6 +259,29 @@ def modify_label(self, pr_number, label, repository_url=None, action='add'):
             return False
         return True
 
+    def comment_on_pr(self, pr_number, content, repository_url=None):
+        api_url = GitHub.api_url(repository_url)
+        if not api_url:
+            return False
+
+        comment_url = '{}/issues/{}/comments'.format(api_url, pr_number)
```
Comment (aj062) [inline Tools/CISupport/ews-build/steps.py]: We can use python 3 f-strings here (and in any new code) to make it more readable. EWS code is python3 only (doesn't support python 2).

**Review by aj062 — APPROVED** [review]

**Comment by JonWBedard** [comment]
Landed https://commits.webkit.org/248505@main (7b036cde6286)!

**Pull request merged**